sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/QueryGenerator.kt

```diff
@@ -177,7 +212,20 @@ abstract class QueryGenerator(
         )
 
         precedingArrays.add(type.name)
-        argumentCounts.add("${type.name}.size")
+        // add list of values to arguments index list
+        // id.mapIndexed { index, _ -> <offset> + <dynamically_calculated_offset> + index*2 + 1 }
+        argumentIndices.add(
```

Comment:
why 2*index? Ideally add a comment or a new named variable

Reply:
The 2*index is used because `createAgument` adds two characters for every argument (for the first '(?' and for every argument after ',?'). I could add an argument to the code generator stating this dependency on the way `createArgument` works and explaining the calculation. This comment won't be in the generated code.

Suggestions:
```
        // the index calculation depends on `createArguments`: every argument requires 2 chars ('(?' or ',?').
        // Therefor we mulitply the index by two to calculate the correct offset
```

Reply:
Thanks for the explanation! I mean adding a new variable in QueryGenerator, but I am also fine with the comment.